daemon/hostdaemon.go

```diff
@@ -89,4 +99,20 @@ func (d *HostDaemon) Start() {
 	}
 	d.addr = addr
 	d.port = port
+
+	server := cniserver.NewCNIServer()
+	err = server.Start()
+
+	if err != nil {
+		d.log.Error(err, "Error starting CNI server for shim")
+	}
+
+	d.cniserver = server
+	go func() {	
+		d.cniserver.Start()
```

Comment:
This code is kind of confusing. We do server.Start() above and we are doing the same server.Start() in a Go routine.

Reply:
Thanks for catching this. It's now resolved as part of the move to `ServeAndListen`.